### PR TITLE
Add support for INSERT in Google Sheets connector

### LINKED ISSUE #3866: Support INSERT in Google Sheets connector
Currently, read operations work as expected. 
Need ability to insert into a gsheet or table.

### PR DESCRIPTION
**Pull request by parambole**
- Added Support for Write Commamnds in Google Sheets connector
- Added Column Type mapping

Fixes #3866

### CODE REVIEW

**Comment by martint** [comment]
@cla-bot check

File: presto-google-sheets/src/main/java/io/prestosql/plugin/google/sheets/SheetsMetadata.java [hunk 1]
```diff
@@ -148,6 +161,55 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
...
+    public void createTable(ConnectorSession session, ConnectorTableMetadata tableMetadata, boolean ignoreExisting)
+    {
+        List<SheetsColumnHandle> columns = buildColumnHandles(tableMetadata);
+        sheetsClient.createSheet(tableMetadata.getTable().getTableName(), columns);
+    }
+
+    @Override
+    public ConnectorOutputTableHandle beginCreateTable(ConnectorSession session, ConnectorTableMetadata tableMetadata, Optional<ConnectorNewTableLayout> layout)
+    {
+        List<SheetsColumnHandle> columns = buildColumnHandles(tableMetadata);
```
Comment (puneetjaiswal) [inline presto-google-sheets/src/main/java/io/prestosql/plugin/google/sheets/SheetsMetadata.java:188]: these two lines are same in `createTable`. can you call `createTable` from here?

File: presto-google-sheets/src/main/java/io/prestosql/plugin/google/sheets/SheetsMetadata.java [hunk 2]
```diff
@@ -148,6 +161,55 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
...
+    public Optional<ConnectorOutputMetadata> finishInsert(ConnectorSession session, ConnectorInsertTableHandle insertHandle, Collection<Slice> fragments, Collection<ComputedStatistics> computedStatistics)
+    {
+        return Optional.empty();
+    }
+
+    @Override
+    public void createTable(ConnectorSession session, ConnectorTableMetadata tableMetadata, boolean ignoreExisting)
+    {
+        List<SheetsColumnHandle> columns = buildColumnHandles(tableMetadata);
+        sheetsClient.createSheet(tableMetadata.getTable().getTableName(), columns);
```
Comment (puneetjaiswal) [inline presto-google-sheets/src/main/java/io/prestosql/plugin/google/sheets/SheetsMetadata.java:182]: what happens if the table is already present? seems `ignoreExisting` is not used.

File: presto-google-sheets/src/test/java/io/prestosql/plugin/google/sheets/TestGoogleSheets.java [hunk 3]
```diff
@@ -45,31 +55,30 @@ protected QueryRunner createQueryRunner()
...
-                    "metadata-sheet-id", TEST_METADATA_SHEET_ID,
-                    "sheets-data-max-cache-size", "1000",
-                    "sheets-data-expire-after-write", "5m"));
+            queryRunner.createCatalog(GOOGLE_SHEETS, GOOGLE_SHEETS, new ImmutableMap.Builder<String, String>()
+                    .put("credentials-path", getTestCredentialsPath())
+                    .put("metadata-sheet-id", "foo_bar_sheet_id#Sheet1")
+                    .put("sheets-data-max-cache-size", "2000")
+                    .put("sheets-data-expire-after-write", "10m")
+                    .put("sheets-range","Sheet1")
+                    .put("sheets-value-input-option","RAW")
```
Comment (puneetjaiswal) [inline presto-google-sheets/src/test/java/io/prestosql/plugin/google/sheets/TestGoogleSheets.java:64]: what are other options except `RAW`? is there a default applied when this property is not specified?

File: presto-google-sheets/src/test/java/io/prestosql/plugin/google/sheets/TestGoogleSheets.java [hunk 4]
```diff
@@ -45,31 +55,30 @@ protected QueryRunner createQueryRunner()
...
+                    .put("drive-permission-type","user")
```
Comment (puneetjaiswal) [inline presto-google-sheets/src/test/java/io/prestosql/plugin/google/sheets/TestGoogleSheets.java:65]: when a new sheet created (using create table command), how is it shared with (or visible to) any other user?

**Review by puneetjaiswal — COMMENTED** [review]
gave it a quick scan, mostly looks good.

File: presto-google-sheets/src/main/java/io/prestosql/plugin/google/sheets/SheetsClient.java [hunk 5]
```diff
@@ -184,12 +227,28 @@ public SheetsClient(SheetsConfig config, JsonCodec<Map<String, List<SheetsTable>
             if (data.get(i).size() >= 2) {
                 String tableId = String.valueOf(data.get(i).get(0));
                 String sheetId = String.valueOf(data.get(i).get(1));
+                if (data.get(i).size() >= 3 && isArchived(String.valueOf(data.get(i).get(3)))) {
```
Comment (chliang71) [inline presto-google-sheets/src/main/java/io/prestosql/plugin/google/sheets/SheetsClient.java:230]: if date.get(i).size() == 3 exactly, then data.get(i).get(3) would actually fail right? the max index is 2 in this case. similarly for the >= 4 below

**Review by chliang71 — COMMENTED** [review]
@parambole thanks for working on this! any updates? Also, would be great if you can share an example google sheet, seems to me more meta data is required in the meta data sheet for this change?

**Comment by fabform** [comment]
Handy to [find google sheet id](https://fabform.io/a/how-to-find-google-sheet-id)

**Comment by amalakar** [comment]
Hi @parambole this would be a very useful feature for users, any update on this?

**Comment by colebow** [comment]
:wave: @parambole - this PR is inactive and doesn't seem to be under development. If you'd like to continue work on this at any point in the future, feel free to re-open.

cc @ebyhr, per your comment on #14356 in case you or someone else wants to pick up this work.

**Pull request closed without merging**